### PR TITLE
Feline Symptom for Uplink: Viro Now Has A Meme TC Buy like the cool kids do

### PR DESCRIPTION
**Pull request by DatBoiTim**
## About The Pull Request
Adds an Uplink Only Symptom and Bottle that contains a gimped version of the fabled cat virus.

Thresholds:
Transmission 12: Similar to Pierrot's Throat, you have a low chance to spread when force speeches.
Stage Speed 12 and Resistance 10: Full-fledged Cat Virus.(Both Must be met for this)

## Why It's Good For The Game
It gives Viro/CMO tots a nice funky but useless TC Buy, and I think having bragging rights TC buys that are role restricted are pretty fun. (Funny Cat Virus Still Go BRRRRR)

## Changelog
:cl: DatBoiTim
add: Added A Feline Symptom, only obtainable through uplink
add: Added an Uplink Entry for the Culture Bottle, restricted to Viro and CMO
/:cl:

### CODE REVIEW

**Comment by NickEmbased** [comment]
speedmerge

**Comment by Archanial** [comment]
Felinid virus already exists.

File: code/datums/diseases/advance/symptoms/feline.dm [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		full_feline = TRUE //OH NYOOO WE NEED TO LYNCH THE VIWOLOGIST
+
+/datum/symptom/feline/Activate(datum/disease/advance/A)
+	if(!..())
+		return
+	var/mob/M = A.affected_mob
+	switch(A.stage)
+		if(2)
+			if(prob(40))
+				M.say(pick("Nya!", "Nya~", "~meow~", "N-nya"))
```
Comment (Archanial) [inline code/datums/diseases/advance/symptoms/feline.dm:40]: Kill me

File: code/datums/diseases/advance/symptoms/feline.dm [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+				to_chat(M, "<span class='danger'>You feel [pick("very catlike", "like avoiding chocolate entirely", "a constant urge to chase after lasers")].</span>")
+		if(4, 5)
+			if(prob(50))
+				M.say(pick("Nya!", "Nya~", "~meow~", "N-nya", "MIAOW!", "N-NYAAAAA!", "NYAAAAAAA~", "Mew - NYAAAAAAA!~", "N- meow", "M- NYAAAA"))
+			if(prob(50))
+				to_chat(M, "<span class='danger'>You feel [pick("incredibly catlike", "like eating chocolate would kill you", "an insatiable urge to chase after lasers", "joining a metagang")].</span>")
+			if(A.stage == 5)
+				if(nya_transmit)
+					if(prob(15))
+						M.say(pick("Nya!", "Nya~", "~meow~", "N-nya", "MIAOW!", "N-NYAAAAA!", "NYAAAAAAA~", "Mew - NYAAAAAAA!~", "N- meow", "M- NYAAAA"))
```
Comment (Archanial) [inline code/datums/diseases/advance/symptoms/feline.dm:56]: Repeated code

File: code/datums/diseases/advance/symptoms/feline.dm [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+						A.spread(5)
+				if(full_feline)
+					if(ishuman(M))
+						M.reagents.add_reagent_list(list(/datum/reagent/mutationtoxin/felinid = 1, /datum/reagent/medicine/mutadone = 1))
```
Comment (Archanial) [inline code/datums/diseases/advance/symptoms/feline.dm:60]: I am pretty sure there is better way to change someone's race without adding chemicals to them

**Review by Archanial — COMMENTED** [review]
Super funny felinids funny hahaha cats cats cope sethe cope harder!!,!

File: code/datums/diseases/advance/symptoms/feline.dm (hunk 3 above)
Comment (DatBoiTim) [reply code/datums/diseases/advance/symptoms/feline.dm:60]: That's how the actual felinid virus does it

File: code/datums/diseases/advance/symptoms/feline.dm (hunk 2 above)
Comment (DatBoiTim) [reply code/datums/diseases/advance/symptoms/feline.dm:56]: It's meant to be.

File: code/datums/diseases/advance/symptoms/feline.dm (hunk 1 above)
Comment (DatBoiTim) [reply code/datums/diseases/advance/symptoms/feline.dm:40]: Yes

**Comment by DatBoiTim** [comment]
> Felinid virus already exists.

It's a derivative. Basically so Viro can TC buy and attempt to make some variant of cat virus.

File: code/datums/diseases/advance/symptoms/feline.dm (hunk 1 above)
Comment (park66665) [reply code/datums/diseases/advance/symptoms/feline.dm:40]: nyannyan uwu

File: code/datums/diseases/advance/symptoms/feline.dm (hunk 2 above)
Comment (DatBoiTim) [reply code/datums/diseases/advance/symptoms/feline.dm:56]: The idea is that the probability of forced speech gets higher with higher stages, and the amount of unique forced speech options also increases.

This is just so, unlike the Pierrot threshhold, there is an actual forcespeech before the spread.

**Comment by ghost** [comment]
This is a terrible idea. speedmerge it!

File: code/datums/diseases/advance/advance.dm [hunk 4]
```diff
@@ -280,8 +280,10 @@
 			severity = DISEASE_SEVERITY_DANGEROUS
 		if(5)
 			severity = DISEASE_SEVERITY_BIOHAZARD
-		if(6 to INFINITY)
+		if(6 to INFINITY && !99)
```
Comment (francinum) [inline code/datums/diseases/advance/advance.dm]: I am like 90% sure this doesn't work. Have you tested this?

**Comment by francinum** [comment]
Why is it it's own severity class for no reason. That entire part is also a mess of less than readable code.

**Comment by DatBoiTim** [comment]
> Why is it it's own severity class for no reason.

Last time I saw the cat virus there was a centcom announcement that a class f biohazard was detected and the nomenclature kinda stuck with me.

**Comment by super12pl** [comment]
Cat virus is admin only for a reason

**Comment by DatBoiTim** [comment]
> Cat virus is admin only for a reason

Power to the Players! GameStop ™️

**Comment by ghost** [comment]
~~Final stage should be turning the victims into cats.~~

This is meh game design.

**Comment by DatBoiTim** [comment]
> ~~Final stage should be turning the victims into cats.~~
> 
> 
> 
> This is meh game design.

Silence you love it

**Comment by ghost** [comment]
> Silence you love it

Only thing I somewhat like here is the thing I half-assedly made. While you're at it, actually, an apid virus would be cooler.

**Comment by DatBoiTim** [comment]
> > Silence you love it
> 
> 
> 
> Only thing I somewhat like here is the thing I half-assedly made. While you're at it, actually, an apid virus would be cooler.
> 
> 

I'll consider it in a future PR

**Comment by DatBoiTim** [comment]
> Why is it it's own severity class for no reason. That entire part is also a mess of less than readable code.

Yeah I am reverting that commit

File: code/datums/diseases/advance/advance.dm (hunk 4 above)
Comment (DatBoiTim) [reply code/datums/diseases/advance/advance.dm]: It does not

File: code/datums/diseases/advance/symptoms/feline.dm (hunk 1 above)
Comment (DatBoiTim) [reply code/datums/diseases/advance/symptoms/feline.dm:40]: Not sure what to do with the review here.

**Comment by utilbird** [comment]
cat virus already exists xd

**Comment by DatBoiTim** [comment]
> cat virus already exists xd
> 
> 

As admin only. Viro has no entertaining meme option for uplink items restricted to them. The pre-baked viruses suck, and lack entertainment value(by either being unfun or extremely mediocre). The experimental diseases are literally TC purchased versions of a cargo item.

I want viro to have a choice to do something funny when they have an uplink.

File: code/datums/diseases/advance/symptoms/feline.dm [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	if(A.properties["stage_speed"] >= 12 && A.properties["resistance"] >= 10)
+		severity+= 99 //Sets Severity to a Custom Felinid Virus Biohazard Classification
+	..()
+
+/datum/symptom/feline/Start(datum/disease/advance/A)
+	if(!..())
+		return
+	if(A.properties["transmittable"] >= 12)
+		nya_transmit = TRUE
+	if(A.properties["stage_speed"] >= 12 && A.properties["resistance"] >= 10)
```
Comment (zeskorion) [inline code/datums/diseases/advance/symptoms/feline.dm:30]: don't just toggle a species change. add a chocolate vulnerability trait and cat organs instead- Species changes, outside of viromerol, are absent from virology intentionally, as species is generally important for making your virus, and the species change proc is dumb in general

**Review by zeskorion — CHANGES_REQUESTED** [review]

File: code/datums/diseases/advance/symptoms/feline.dm (hunk 5 above)
Comment (DatBoiTim) [reply code/datums/diseases/advance/symptoms/feline.dm:30]: ~~This was a meme PR~~ I literally took the species change method from the cat virus.

Comment (zeskorion) [reply code/datums/diseases/advance/symptoms/feline.dm:30]: that's an adminbus only disease, not something obtainable ingame

**Comment by DatBoiTim** [comment]
> > cat virus already exists xd
> > 
> > 
> 
> As admin only. Viro has no entertaining meme option for uplink items restricted to them. The pre-baked viruses suck, and lack entertainment value(by either being unfun or extremely mediocre). The experimental diseases are literally TC purchased versions of a cargo item.
> 
> I want viro to have a choice to do something funny when they have an uplink.
> 
> 

Okay, I need to prune my branches and cut down on open PRs. This was a meme PR. I've had my fun with it. But I've got multiple other projects that will take precedence over "Hahaha forced felinid disease funee"

**Pull request closed without merging**